apis/v1alpha1/ansibleRun_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// A ConfigurationSource represents the source of a AnsibleRun Configuration.
+// +kubebuilder:validation:Enum=Remote;Inline
+type ConfigurationSource string
+
+// Module sources.
+const (
+	ConfigurationSourceRemote ConfigurationSource = "Remote"
+	ConfigurationSourceInline ConfigurationSource = "Inline"
```

Comment:
For the configuration source, my understanding is that the two options, remote and inline, are now mutual exclusive. I am thinking whether it needs to be changed. The reason for that:

I think the remote option would be more useful in real projects, as usually people will run Ansible automation using a bunch of well-organized yaml stuff, rather than a single playbook.yml file. The inline option would be more likely for demo purpose.

However, I am thinking one use case that probably can lead us to have both remote and inline at the same time. When call one or more roles as reusable building blocks from somewhere such as git repo or ansible galaxy, people often create playbook on top of that w/ some additional twists. This playbook may not be part of the existing contents in repos, but instead, is created on the fly, which can nicely implemented by the inline option.

Does it make sense?

Reply:
Yes this makes sense. this Can be a future improvement. Would you please open a ticket with this concern.

Reply:
Sure, I just opened an issue at: https://github.com/multicloudlab/crossplane-provider-ansible/issues/26.

Reply:
Thanks @morningspace